school_evaluation/models/school_evaluation.py

```diff
@@ -35,9 +21,47 @@ def _compute_total_points(self):
                 rec.total = sum(line.point_id.rating for line in rec.eval_line
                                 if line.point_id.rating)
 
+    student_id = fields.Many2one('student.student', 'Student Name',
+                                 help="Select Student")
+    teacher_id = fields.Many2one('school.teacher', "Teacher",
+                                 help='Select teacher')
+    type = fields.Selection([('student', 'Student'),
+                             ('faculty', 'Faculty')],
+                            'User Type', required=True,
+                            help="Type of evaluation")
+    date = fields.Date('Evaluation Date', required=True,
+                       help="Evaluation Date",
+                       default=fields.Date.context_today)
+    eval_line = fields.One2many('school.evaluation.line', 'eval_id',
+                                'Questionnaire',
+                                help='Enter evaluation details')
+    total = fields.Float('Total Points', compute='_compute_total_points',
+                         method=True, help="Total Points Obtained",
+                         store="True")
+    state = fields.Selection([('draft', 'Draft'), ('start', 'In Progress'),
+                              ('finished', 'Finish'), ('cancelled', 'Cancel')],
+                             'State', readonly=True, default='draft',
+                             help='State of evaluation line')
+    username = fields.Many2one('res.users', 'User', readonly=True,
+                               default=lambda self: self.env.user,
+                               help='Related user')
+    active = fields.Boolean('Active', default=True,
+                            help='Activate/Deactivate record')
+
+    @api.model
+    def default_get(self, fields):
+        '''Override method to get default value of teacher'''
+        res = super(SchoolEvaluation, self).default_get(fields)
+        if res.get('type') == 'student':
+            hr_emp_rec = self.env['school.teacher'].search([
+                                        ('user_id', '=', self._uid)])
+            res.update({'teacher_id': hr_emp_rec.id})
```

Comment:
If there is a case we didn't find any record in hr_emp_rec, hr_emp_rec.id will raise traceback.

Reply:
Yeas ma'am, have checked, It won't give any error as if that would be the case it consider the null value itself.

Reply:
It should update everywhere, @AnuKPatel @JayVora-SerpentCS what's your opinion in this?